sound/soc/sof/debug.c

```diff
@@ -162,12 +164,12 @@ int snd_sof_dbg_init(struct snd_sof_dev *sdev)
 		err = snd_sof_debugfs_io_create_item(sdev, sdev->bar[map->bar] +
 						  map->offset, map->size,
 						  map->name);
+		/* errors are only due to memory allocation, not debugfs */
 		if (err < 0)
-			dev_err(sdev->dev, "error: cannot create debugfs for %s\n",
-				map->name);
+			return err;
```

Comment:
@plbossart this patch is fine but we have plenty of other places where we call snd_sof_debugfs_io_create_item and do not check for errors even if it fails due to mem alloc. Should we modify those too?

Reply:
@ranj063 Yes, I saw that, but it looks like a larger change since there is no return status for e.g. bdw_get_info, and if memory allocation fails at this point I am not sure what happens. The changes would be in Intel-specific parts and we can deal with this later.